### PR TITLE
[americastestkitchen] Add support for downloading entire series

### PR DESCRIPTION
**Pull request by bm1549**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

This PR adds the ability to download entire series of ATK as youtube-dl playlists.

### CODE REVIEW

**Review by lilykuong — APPROVED** [review]

File: youtube_dl/extractor/americastestkitchen.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def _real_extract(self, url):
 
 
 class AmericasTestKitchenSeasonIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?(?P<show>americastestkitchen|cookscountry)\.com/episodes/browse/season_(?P<id>\d+)'
+    _VALID_URL = r'https?:\/\/(?:www\.)?(?P<show>americastestkitchen|cookscountry)\.com\/?(episodes\/browse\/season_(?P<season>\d+))?'
```
Comment (dirkf) [inline youtube_dl/extractor/americastestkitchen.py:97]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?(?P<show>americastestkitchen|cookscountry)\.com(?:/(?:episodes/browse/season_(?P<season>\d+))?)?'
```

File: youtube_dl/extractor/americastestkitchen.py [hunk 2]
```diff
@@ -111,29 +111,49 @@ class AmericasTestKitchenSeasonIE(InfoExtractor):
...
+        slug, title = ('atk', 'America\'s Test Kitchen') if show == 'americastestkitchen' else ('cco', 'Cook\'s Country')
 
-        season = 'Season %d' % season_number
+        facet_filters = [
+            'search_document_klass:episode',
+            'search_show_slug:' + slug,
+        ]
+
+        if season_number:
+            facet_filters.append('search_atk_episode_season:' + season)
```
Comment (dirkf) [inline youtube_dl/extractor/americastestkitchen.py:146]: I had to use this:
```suggestion
            facet_filters.append('search_season_list:' + season)
```

File: youtube_dl/extractor/americastestkitchen.py [hunk 3]
```diff
@@ -145,15 +165,17 @@ def entries():
...
                     'title': episode.get('title'),
                     'description': episode.get('description'),
                     'timestamp': unified_timestamp(episode.get('search_document_date')),
-                    'season_number': season_number,
+                    'season_number': episode.get('search_atk_episode_season'),
                     'episode_number': int_or_none(episode.get('search_%s_episode_number' % slug)),
                     'ie_key': AmericasTestKitchenIE.ie_key(),
                 }
 
+        playlist_id, playlist_title = ('season_%d' % season_number, 'Season %d' % season_number) if season_number else (show, title)
```
Comment (dirkf) [inline youtube_dl/extractor/americastestkitchen.py:178]: Could merge this with the conditional block at l.145.

**Review by dirkf — CHANGES_REQUESTED** [review]
Thanks for your work!

I made a few suggestions but, good news, I wanted to test them and ended up with the attached [americastestkitchen.py](https://github.com/ytdl-org/youtube-dl/files/10501697/americastestkitchen.py.txt) that passes all the old and new tests and pulls in anything useful that yt-dlp had added. By all means, if you're happy with my version, commit it into your branch instead of responding to the suggestions.

File: youtube_dl/extractor/americastestkitchen.py [hunk 4]
```diff
@@ -105,48 +121,93 @@ class AmericasTestKitchenSeasonIE(InfoExtractor):
...
+            facet_filters.append('search_season_list:' + playlist_title)
+        else:
+            playlist_id = show
+            playlist_title = title
 
         season_search = self._download_json(
             'https://y1fnzxui30-dsn.algolia.net/1/indexes/everest_search_%s_season_desc_production' % slug,
-            season, headers={
-                'Origin': 'https://www.%s.com' % show_name,
+            playlist_title, headers={
```
Comment (dirkf) [inline youtube_dl/extractor/americastestkitchen.py:191]: Prompted by your variable rename, I realised that this parameter is meant to be an ID. Let's use that instead.
```suggestion
            playlist_id, headers={
```

**Review by dirkf — APPROVED** [review]
I made a small update and otherwise good to go, thanks.

**Comment by bm1549** [comment]
@dirkf could you merge this in for me? I don't have the permissions to do it

**Pull request merged**